src/node_shadow_realm.cc

```diff
@@ -32,6 +40,10 @@ MaybeLocal<Context> HostCreateShadowRealmContextCallback(
     Local<Context> initiator_context) {
   Environment* env = Environment::GetCurrent(initiator_context);
   EscapableHandleScope scope(env->isolate());
+
+  // We do not expect the realm bootstrapping to throw any
+  // exceptions. If it does, exit the current Node.js instance.
+  TryCatchScope try_catch(env, TryCatchScope::CatchMode::kFatal);
```

Comment:
Note that `TryCatchScope` uses enhancers from the main context...as we are refactoring a bunch of env access to realms already, it might be safer to refactor that and especially `ReportFatalException`.

Reply:
This `TryCatchScope` here is only used when the built-in modules throw errors in development. Yet it is true that it should be refactored to enhance the error in a specific realm since it calls into JavaScript. I'll do this in a follow-up PR.

Reply:
okay, I am okay with a follow-up, just seeing that there are already a bunch of env -> realm refactorings, wondered if you want to just do it together here.